src/Silex/ContentNegotiationProvider.php

```diff
@@ -0,0 +1,104 @@
+    const ATTRIBUTE_ACCEPT_CHARSET = '_accept_charset';
+    const ATTRIBUTE_ACCEPT_ENCODING = '_accept_encoding';
+    const ATTRIBUTE_ACCEPT_LANGUAGE = '_accept_language';
+
+    public function register(Container $app)
+    {
+        $app['negotiator.accept'] = function () {
+            return new Negotiator();
+        };
+
+        $app['negotiator.accept_charset'] = function () {
+            return new CharsetNegotiator();
+        };
+
+        $app['negotiator.accept_encoding'] = function () {
+            return new EncodingNegotiator();
+        };
+
+        $app['negotiator.accept_language'] = function () {
+            return new LanguageNegotiator();
+        };
+
+        $app['content_negotiator.accept'] = function () use ($app) {
+            return new ContentNegotiator($app['negotiator.accept'], 'Accept', self::ATTRIBUTE_ACCEPT);
+        };
+
+        $app['content_negotiator.accept_charset'] = function () use ($app) {
+            return new ContentNegotiator($app['negotiator.accept_charset'], 'Accept-Charset', self::ATTRIBUTE_ACCEPT_CHARSET);
+        };
+
+        $app['content_negotiator.accept_encoding'] = function () use ($app) {
+            return new ContentNegotiator($app['negotiator.accept_encoding'], 'Accept-Encoding', self::ATTRIBUTE_ACCEPT_ENCODING);
+        };
+
+        $app['content_negotiator.accept_language'] = function () use ($app) {
+            return new ContentNegotiator($app['negotiator.accept_language'], 'Accept-Language', self::ATTRIBUTE_ACCEPT_LANGUAGE);
+        };
+
+        $app['negotiate.accept'] = function () {
+            return function (string ...$types) {
+                return function (Request $request, Container $app) use ($types) {
+                    return $app['content_negotiator.accept']->negotiate($request, $types);
+                };
+            };
+        };
+
+        $app['negotiate.accept_charset'] = function () {
+            return function (string ...$types) {
+                return function (Request $request, Container $app) use ($types) {
+                    return $app['content_negotiator.accept_charset']->negotiate($request, $types);
+                };
+            };
+        };
+
+        $app['negotiate.accept_encoding'] = function () {
+            return function (string ...$types) {
+                return function (Request $request, Container $app) use ($types) {
+                    return $app['content_negotiator.accept_encoding']->negotiate($request, $types);
+                };
+            };
+        };
+
+        $app['negotiate.accept_language'] = function () {
+            return function (string ...$types) {
+                return function (Request $request, Container $app) use ($types) {
+                    return $app['content_negotiator.accept_language']->negotiate($request, $types);
+                };
+            };
+        };
+
+        if (interface_exists(ArgumentValueResolverInterface::class)) {
```

Comment:
what's the use case in which this interface doesn't exist? I guess old versions of `symfony/http-kernel` from the tests

Reply:
Added comment.